### PR TITLE
Login Screen using AuthContext

### PR DESCRIPTION
**Pull request by FranRossi**
_Resolve_: [closes -#865cbq5mm](https://app.clickup.com/t/865cbq5mm)

## Description ⭐

I've added AuthContext and a Login Screen so the user can enter proper credentials to use the app.

### Type of change 🏃

- [ ] Bug fix (non-breaking change which fixes an issue)
- ✅ New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots 📷

image

### CODE REVIEW

File: screens/Register/RegisterScreen.tsx [hunk 1]
```diff
@@ -125,7 +129,7 @@ export default function RegisterScreen({
         </TouchableOpacity>
         <View style={styles.loginPageContainer}>
           <Text style={[styles.loginText]}>Already have an account?</Text>
-          <TouchableOpacity onPress={() => navigation.navigate("Home")}>
+          <TouchableOpacity onPress={() => navigation.navigate("Login")}>
```
Comment (ddoumecq) [inline screens/Register/RegisterScreen.tsx:132]: if you put `Foo` here, does typescript give you an error? if it doesn't we either need to configure this better OR we need to put the screen names in a constant

File: package.json [hunk 2]
```diff
@@ -14,10 +14,11 @@
     "axios": "^1.4.0",
     "expo": "~48.0.15",
     "expo-checkbox": "~2.3.1",
+    "expo-secure-store": "~12.1.1",
     "expo-status-bar": "~1.4.4",
     "react": "18.2.0",
     "react-native": "0.71.7",
-    "react-native-gesture-handler": "^2.10.0",
```
Comment (ddoumecq) [inline package.json:20]: interesting, why did you need to downgrade?

File: context/AuthProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import React, { createContext, useState } from "react";
+import axiosConfig, { setUserToken } from "../helpers/axios/axiosConfig";
```
Comment (ddoumecq) [inline context/AuthProvider.tsx]: We normally just call the global  axios object `axios`
```suggestion
import axios, { setUserToken } from "../helpers/axios/axiosConfig";
```

File: Root.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      {user ? (
+        <NavigationContainer>
+          <HomeStackNavigator />
+        </NavigationContainer>
+      ) : (
+        <NavigationContainer>
+          <AuthStackNavigator />
+        </NavigationContainer>
+      )}
+    </>
```
Comment (ddoumecq) [inline Root.tsx:121]: it would be easier to follow to format this code as:

```
  <NavigationContainer>
      {user ? (
          <HomeStackNavigator />
      ) : (
          <AuthStackNavigator />
      )}
  </NavigationContainer>
```

**Review by ddoumecq — APPROVED** [review]

File: screens/Register/RegisterScreen.tsx (hunk 1 above)
Comment (FranRossi) [reply screens/Register/RegisterScreen.tsx:132]: Yes It does give me an error if I switch Login for Foo, it must be declared on the ParamList type

File: package.json (hunk 2 above)
Comment (FranRossi) [reply package.json:20]: It was something expo suggested me, it said some things would not work properly if I use that version and suggested me which one to install instead

File: context/AuthProvider.tsx (hunk 3 above)
Comment (FranRossi) [reply context/AuthProvider.tsx]: Thanks for the suggestion, done!

File: Root.tsx (hunk 4 above)
Comment (FranRossi) [reply Root.tsx:121]: Yes, that makes sense, always leave the NavigationContainer no matter which case I am rendering
